client/src/types/OnDeviceComponentRequest.ts

```diff
@@ -66,6 +72,103 @@ export namespace ODC {
 		responseMaxChildDepth?: number;
 	}
 
+	export interface RequestOptions {
+		/** How long to wait (in milliseconds) until the request is considered a failure. If not provided OnDeviceComponent.defaultTimeout is used  */
+		timeout?: number;
+	}
+
+	export interface Request {
+		id: string;
+		args: RequestArgs;
+		type: RequestTypes;
+		settings: {
+			logLevel: LogLevels
+		};
+		callback?: (response: ODC.RequestResponse) => void;
+	}
+
+	export interface RequestResponse {
+		json: any;
+		stringLength: number;
+		binaryLength: number;
+		stringPayload: string;
+		binaryPayload: Buffer;
+	}
+
+	export interface NodeRepresentation {
+		// Allow other fields
+		[key: string]: any;
+		change: {
+			Index1: number;
+			Index2: number;
+			Operation: string;
+		};
+		childRenderOrder?: 'renderFirst' | 'renderLast';
+		children: NodeRepresentation[];
+		clippingRect?: [number, number, number, number];
+		enableRenderTracking?: boolean;
+		focusedChild: NodeRepresentation;
+		focusable: boolean;
+		id: string;
+		inheritParentOpacity?: boolean;
+		inheritParentTransform?: boolean;
+		muteAudioGuide?: boolean;
+		opacity?: number;
+		renderPass?: number;
+		renderTracking?: 'none' | 'partial' | 'full';
+		rotation?: number;
+		scale?: [number, number];
+		scaleRotateCenter?: [number, number];
+		subtype: string;
+		translation?: [number, number];
+		visible?: boolean;
+	}
+
+	export interface NodeTree {
```

Comment:
I don't have full understanding of all these things, but calling this "tree" doesn't feel right. a "tree" would be the entire thing in my opinion. What about:
 - `TreeNode`
 - `SGNode`

Reply:
I hadn't thought of `TreeNode` before. Definitely get the confusion. I had thought of `NodeBranch` before but that gets confusing as even the base of the tree would still be a branch. The other two options don't work well as in the code we often have the actual node used alongside the tree object so naming variables there would be difficult.

Reply:
After talking, will move forward with using `TreeNode`. Will be handled in a separate PR.